controllers/volumereplicationgroup_controller.go

```diff
@@ -1141,10 +1139,18 @@ func (v *VRGInstance) undoPVRetentionForPVC(pvc corev1.PersistentVolumeClaim, lo
 	return nil
 }
 
-// Upload PV to the list of S3 stores in the VRG spec
-func (v *VRGInstance) uploadPVToS3Profiles(pvc corev1.PersistentVolumeClaim) (err error) {
+// protectPVClusterDatauploads the PV related cluster data of the given
+// PVC to the list of S3 stores in the VRG spec
+func (v *VRGInstance) protectPVClusterData(pvc *corev1.PersistentVolumeClaim,
+	log logr.Logger) (err error) {
 	for _, s3ProfileName := range v.instance.Spec.S3ProfileList {
 		if err := v.uploadPV(s3ProfileName, pvc); err != nil {
+			msg := "Failed to upload PV cluster data to s3Profile " + s3ProfileName
+			v.updateProtectedPVCCondition(pvc.Name, PVCError, msg)
```

Comment:
We need to split this into 2 conditions. I know this is prep PR, but we should avid fiddling conditions in here, and handle it at a higher layer (afaict).

Reply:
I don't believe you're suggesting to add the condition in the caller, `protectPVCsAsPrimary` that calls three other methods, each of which add conditions inside them. Hence, I thought  it will be consistent to add conditions inside this method.

Reply:
Are you starting with a condition per PVC or a global VRG condition (I would assume the latter to get the feature going), if the former then the code churn would be larger.

Reply:
Currently, there is an `Available` condition per PVC, which is false during `progressing` or `error` states. When there's a PV upload error, we could set the corresponding PVC condition to error with a msg indicating the reason. This PVC error condition gets folded up into global VRG condition as error. This is also in line with Raghu suggests: use only one condition per PVC.

Thus the VRG (global) condition will be in error either due to lack of PV backend data protection or PV cluster data protection. 

DRPC should allow a relocate action only if it sees that the VRG (global) condition is available. DRPC should proceed with the failover action requested by the user even if the VRG is not in available condition.

Reply:
Ok, so we are modelling this as per-PVC and roll-up to a VRG condition. That works, thanks.